src/robot_module/doc/lock_service.puml

```diff
@@ -0,0 +1,27 @@
+@startuml Locking Service Sequence Diagram

```

Comment:
PUML is a diagram generator we use at Microsoft, it's free and is a nice way to create pictures from text.

You can generate the diagram by copying this into the [bottom of the plant uml website](https://plantuml.com/) or downloading a local viewer (there's a vs code extension called PlantUml by Jebbs that has some instructions on that)

Also here's a [link to the image](https://www.plantuml.com/plantuml/png/TP51JyCm38Nl-HK-me7zW1tG9d51ue0qSKEQJEfLrbod9Cwc_NkSGISAwgdjyyltNfrDCSnM1md69WGlAHvPUzXHFd4aVtvMKdzvOkmp3c5qEqSUKGssrT9-5pEF1bW0_NCsVtjkSkMPs1Bs4zx7KaAG0W-FReIr1sfNW1G_X62WKh0dI2VAO6da6C9gjGgXHRKZKcv74CJ9a5lXOd-Hq75PGZQhUnjpSbzHCUc1yu36INzoT-HYj__ru_CMymwPh6Rz9XwGfUPR_WBfl-haLygSEeueSe6HjNzNRshv0kZWdkoQkhIPgP8GbblyL6qDstZKT1RgUfexxpI14dNWl3EotL_HywKkNzF6ntMGBm00)